app/gui/src/dashboard/components/AriaComponents/Form/components/useForm.ts

```diff
@@ -76,6 +76,10 @@ export function useForm<Schema extends types.TSchema, SubmitResult = void>(
 
     const computedSchema = typeof schema === 'function' ? schema(schemaModule.schema) : schema
 
+    // We need to disable the eslint rules here, because we call hooks conditionally
+    // but it's safe to do so, because we don't switch between the two types of arguments
+    // and if we do, we throw an error.
+    // eslint-disable-next-line react-compiler/react-compiler
```

Comment:
Could you split the component into two variants instead? That would also remove the invariant form code.

Reply:
whoops... i don't think it needs to take a union at all.

(also this is a hook not a component. the path may be a misnomer...)